### PR TITLE
Speed up parallel plugin setup #310

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

File: daemon/src/main/java/org/mvndaemon/mvnd/plugin/CliPluginDescriptorCache.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+import org.codehaus.plexus.component.repository.ComponentDescriptor;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.WorkspaceRepository;
+import org.eclipse.sisu.Priority;
+import org.eclipse.sisu.Typed;
+
+/*
+ * gnodet: This file is based on maven DefaultPluginDescriptorCache and changed in order
```
Comment (ppalaga) [inline daemon/src/main/java/org/mvndaemon/mvnd/plugin/CliPluginDescriptorCache.java:46]: It would be handy to have a separate commit with the unchanged original file to see what exactly we have changed.

**Review by ppalaga — COMMENTED** [review]
Do we have at least one test (which?) covering this?

**Comment by gnodet** [comment]
> Do we have at least one test (which?) covering this?

Not at the moment.  Not really sure how to test that for now.  Once I'm done with adding transfer events back to the client, it may be easier to test that downloads can be done in parallel.

File: daemon/src/main/java/org/mvndaemon/mvnd/plugin/CliPluginDescriptorCache.java (hunk 1 above)
Comment (gnodet) [reply daemon/src/main/java/org/mvndaemon/mvnd/plugin/CliPluginDescriptorCache.java:46]: Yes, that makes sense.

**Comment by ppalaga** [comment]
> Once I'm done with adding transfer events back to the client, it may be easier to test that downloads can be done in parallel.

Thanks for explaining, no problem with that plan. Because we have a bunch of unreleased features in the master, how about releasing now and have more time for maturing this?

**Comment by gnodet** [comment]
> Because we have a bunch of unreleased features in the master, how about releasing now and have more time for maturing this?

+1
